lib/src/utf8.dart

```diff
@@ -8,54 +8,93 @@ import 'dart:typed_data';
 
 import 'package:ffi/ffi.dart';
 
-/// [Utf8] implements conversion between Dart strings and zero-terminated
-/// UTF-8 encoded "char*" strings in C.
+/// The contents of a native zero-terminated array of UTF-8 code units.
 ///
-/// [Utf8] is represented as a struct so that `Pointer<Utf8>` can be used in
-/// native function signatures.
-//
-// TODO(https://github.com/dart-lang/ffi/issues/4): No need to use
-// 'asTypedList' when Pointer operations are performant.
+/// The Utf8 type itself has no functionality, it's only intended to be used
+/// through a `Pointer<Utf8>` representing the entire array. This pointer is
+/// the equivalent of a char pointer (`const char*`) in C code. The individual
```

Comment:
Is it really `char*`, or is it `unsigned char*`?
C `char`s are not guaranteed to be signed or unsigned, but UTF-8 is definitely unsigned.

Reply:
The signedness of `char` is undefined in C indeed, however most APIs and code bases use `const char*`/`char*` regardless.